File: src/types/FunctionSchema.ts
// Interfaces for FunctionSchema retrieval processes

/**
 * Interface for request to read Project's FunctionSchemas.
 */
export interface ReadProjectFunctionSchemasRequest {
  project_uuid: string;
}

/**
 * Interface for request to read FunctionSchema.
 */
export interface ReadFunctionSchemaRequest {
  uuid: string;
}

/**
 * General interface for FunctionSchema.
 */
export interface FunctionSchema {
  id: number;
  created_at: string;
  uuid: string;
  name: string;
  parameters: Record<string, any>;
  online: boolean;
  description: string;
  mock_response: Record<string, any>;
  project_uuid: string;
}

File: backend/api/web/models/function_schema.py
"""APIs for FunctionSchema"""
from typing import Any, Dict, Optional

from modules.types import PMObject


class FunctionSchemaInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    name: str
    parameters: Dict[str, Any]
    online: bool

    description: Optional[str] = None
    mock_response: Optional[Dict[str, Any]] = None

    project_uuid: str
